src/core/public/chrome/ui/header/nav_link.test.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { isActiveNavLink, createEuiListItem } from './nav_link';
+import { ChromeNavLink } from '../../..';
+import { httpServiceMock } from '../../../http/http_service.mock';
+
+describe('isActiveNavLink', () => {
+  it('should return true if the currentId is "discover" and targetId is "discover"', () => {
+    expect(isActiveNavLink('discover', 'discover')).toBe(true);
+  });
+
+  it('should return true if the currentId is "data-explorer" and targetId is "data-explorer"', () => {
+    expect(isActiveNavLink('data-explorer', 'data-explorer')).toBe(true);
+  });
+
+  it('should return true if the currentId is "discover" and targetId is "data-explorer"', () => {
+    expect(isActiveNavLink('discover', 'data-explorer')).toBe(true);
+  });
```

Comment:
This is a great example of the value of unit tests in capturing developer intent! It helps reviewers understand the intended solution. And actually, I'm not sure this particular outcome is desired - based on the current implementation of `data-explorer` and `discover`, `data-explorer` should never have a nav link, and if we changed our mind about that, we wouldn't actually want this behavior (because both nav links would be marked active). See my suggested refactor below, which would switch the behavior of this case but leave all the others as-is.

Reply:
Ah, thanks @joshuarrrr for pointing out the logic error here! I didn't notice it when I was writing the test. Much appreciated!